include/maliput/routing/route_phase.h

```diff
@@ -0,0 +1,174 @@
+    // TODO(#543): Validate that start_road_position_.lane->id() and position are in the default_lane_s_range and
+    // adjacent_lane_s_ranges.
+    // TODO(#543): Validate that start_road_position_.lane->id() and position are in the default_lane_s_range and
+    // adjacent_lane_s_ranges.
+    // TODO(#543): Validate that adjancent_lane_s_ranges_ are effectively adjacent to default_lane_s_range.
+  }
+
+  /// @return The index of this RoutePhase.
+  int Index() const { return index_; }
+
+  /// @return The start maliput::api::RoadPosition of this RoutePhase.
+  const maliput::api::RoadPosition& StartRoadPosition() const { return start_road_position_; }
+
+  /// @return The end maliput::api::RoadPosition of this RoutePhase.
+  const maliput::api::RoadPosition& EndRoadPosition() const { return end_road_position_; }
+
+  /// @return The default maliput::api::LaneSRange.
+  const maliput::api::LaneSRange& DefaultLaneSRange() const { return default_lane_s_range_; }
+
+  /// @return The maliput::api::rules::RoadRulebook::QueryResults for the
+  /// default maliput::api::LaneSRange.
+  maliput::api::rules::RoadRulebook::QueryResults RulesForDefaultLaneSRange() const {
+    return road_network_->rulebook()->FindRules({default_lane_s_range_});
+  }
+
+  /// @return A vector of maliput::api::LaneSRanges with the adjacent
+  /// maliput::api::LaneSRanges.
+  std::vector<maliput::api::LaneSRange> AdjacentLaneSRanges() const { return adjancent_lane_s_ranges_; }
+
+  /// Finds the rules for @p lane_s_range.
+  ///
+  /// The @p lane_s_range must intersect one of the default or adjacent
+  /// maliput::api::LaneSRanges.
+  /// (To be discussed): we probably want it to be included rather than intersected.
+  ///
+  /// @param lane_s_range The maliput::api::LaneSRange to consider.
+  /// @return The rules that apply to that @p lane_s_range.
+  /// @throws maliput::common::assertion_error When @p lane_s_range does not
+  /// intersect any of the default or adjacent maliput::api::LaneSRanges.
+  maliput::api::rules::RoadRulebook::QueryResults RulesForLaneSRange(
+      const maliput::api::LaneSRange& lane_s_range) const;
+
+  /// Finds the maliput::api::LaneSRange where @p inertial_position falls into.
+  ///
+  /// @param inertial_position An INERTIAL-Frame position.
+  /// @return An optional with the maliput::api::LaneSRange when the RoutePhase
+  /// contains a LaneSRange which holds @p inertial_position.
+  std::optional<maliput::api::LaneSRange> FindLaneSRangeBy(
+      const maliput::api::InertialPosition& inertial_position) const;
+
+  /// Finds the maliput::api::LaneSRange where @p road_position falls into.
+  ///
+  /// @param road_position A LANE-Frame position.
+  /// @return An optional with the maliput::api::LaneSRange when the RoutePhase
+  /// contains a LaneSRange which holds @p road_position.
+  std::optional<maliput::api::LaneSRange> FindLaneSRangeBy(
+      const maliput::api::RoadPosition& road_position) const;
+
+ private:
+  int index_{};
```

Comment:
these can be const

Reply:
I need it to not be const to keep the copy / move / assign semantics.